homeassistant/components/alarm_control_panel/egardia.py

```diff
@@ -94,24 +94,24 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
         _LOGGER.info("Setting up EgardiaServer")
         try:
             if D_EGARDIASRV not in hass.data:
-                hass.data[D_EGARDIASRV] = egardiaserver.EgardiaServer('',
-                                                                      rs_port)
-                bound = hass.data[D_EGARDIASRV].bind()
+                server = egardiaserver.EgardiaServer('', rs_port)
+                hass.data[D_EGARDIASRV] = server
+                bound = server.bind()
 
                 if not bound:
                     raise IOError("Binding error occurred while " +
                                   "starting EgardiaServer")
-                hass.data[D_EGARDIASRV].register_callback(
-                    eg_dev.handle_status_event)
-                hass.data[D_EGARDIASRV].start()
+                server.register_callback(eg_dev.handle_status_event)
```

Comment:
Registering a callback should be done for each `setup_platform` run, so move that out of the check for existing server in `hass.data `. It shouldn't matter much if you put it after `server.start()`.